tests/test_active_surface.py

```diff
@@ -76,6 +76,212 @@ def test_soft_reset(self):
+        returns the byte_ack followed by the driver version
+        expressed as a byte when the message is completed."""
+        msg = b'\xFA\x20\x10'
+        checksum = b'\xD5'
+        for byte in msg:
+            self.assertTrue(self.system.parse(byte))
+        response = self.system.parse(checksum)
+        expected_length = 4  #byte_ack + byte_start + version + byte_checksum
+        self.assertEqual(expected_length, len(response))
+        self.assertEqual(response[0], byte_ack)
+
+    def test_soft_stop(self):
+        msg = b'\xFA\x20\x11'
+        checksum = b'\xD4'
+        for byte in msg:
+            self.assertTrue(self.system.parse(byte))
+        self.assertEqual(self.system.parse(checksum), byte_ack)
+
+    def test_get_position(self):
+        """The system returns True for every proper byte, and
+        returns the byte_ack followed by the current driver position
+        expressed as 4 bytes when the message is completed."""
+        msg = b'\xFA\x20\x12'
+        checksum = b'\xD3'
+        for byte in msg:
+            self.assertTrue(self.system.parse(byte))
+        response = self.system.parse(checksum)
+        expected_length = 7  #byte_ack + byte_start + pos 3-0 + byte_checksum
+        self.assertEqual(expected_length, len(response))
+        self.assertEqual(response[0], byte_ack)
+
+    def test_get_status(self):
+        """The system returns True for every proper byte, and
+        returns the byte_ack followed by the current driver status
+        expressed as 3 bytes when the message is completed."""
+        msg = b'\xFA\x20\x13'
+        checksum = b'\xD2'
+        for byte in msg:
+            self.assertTrue(self.system.parse(byte))
+        response = self.system.parse(checksum)
+        expected_length = 6  #byte_ack + byte_start + status 0-2 + byte_checksum
+        self.assertEqual(expected_length, len(response))
+        self.assertEqual(response[0], byte_ack)
+
+    def test_get_driver_type(self):
+        """The system returns True for every proper byte, and
+        returns the byte_ack followed by the current driver type
+        expressed as a byte when the message is completed."""
+        msg = b'\xFA\x20\x14'
+        checksum = b'\xD1'
+        for byte in msg:
+            self.assertTrue(self.system.parse(byte))
+        response = self.system.parse(checksum)
+        expected_length = 4  #byte_ack + byte_start + driver_type + byte_checksum
+        self.assertEqual(expected_length, len(response))
+        self.assertEqual(response[0], byte_ack)
+
+    def test_set_in_range_min_frequency(self):
+        msg = b'\xFA\x60\x20\x03\xE8'  #Min frequency set to 1000Hz, OK
+        checksum = b'\x9A'
+        for byte in msg:
+            self.assertTrue(self.system.parse(byte))
+        self.assertEqual(self.system.parse(checksum), byte_ack)
+
+    def test_set_out_range_min_frequency(self):
+        msg = b'\xFA\x60\x20\x00\x0A'  #Min frequency set to 10Hz, too low
```

Comment:
In the comment, something like:
`# Set min frequency to 10Hz (\x0A), but min allowed value is 5Hz`.  In this way I can see which byte is the frequency, and also the min value allowed.